src/frontend/komponenter/Fagsak/Vilkårsvurdering/useVilkårsvurderingApi.ts

```diff
@@ -0,0 +1,167 @@
+import { useState } from 'react';
+
+import { useHttp } from '@navikt/familie-http';
+import type { Ressurs } from '@navikt/familie-typer';
+import { RessursStatus } from '@navikt/familie-typer';
+
+import { useBehandling } from '../../../context/behandlingContext/BehandlingContext';
+import type { IBehandling } from '../../../typer/behandling';
+import type {
+    IRestAnnenVurdering,
+    IRestNyttVilkår,
+    IRestPersonResultat,
+    VilkårType,
+} from '../../../typer/vilkår';
+
+export const useVilkårsvurderingApi = () => {
+    const { request } = useHttp();
+
+    const { åpenBehandling, settÅpenBehandling } = useBehandling();
+
+    const behandlingId =
+        åpenBehandling.status === RessursStatus.SUKSESS ? åpenBehandling.data.behandlingId : null;
+
+    const [lagrerVilkår, settLagrerVilkår] = useState<boolean>(false);
+    const [oppretterVilkår, settOppretterVilkår] = useState<boolean>(false);
+    const [sletterVilkår, settSletterVilkår] = useState<boolean>(false);
+    const [lagrerAnnenVurdering, settLagrerAnnenVurdering] = useState<boolean>(false);
+    const [lagrerAnnenVurderingFeilmelding, settLagrerAnnenVurderingFeilmelding] =
+        useState<string>('');
+    const [oppretterVilkårFeilmelding, settOppretterVilkårFeilmelding] = useState<string>('');
+
+    const lagreVilkår = (
+        restPersonResultat: IRestPersonResultat,
+        vilkårId: number,
+        onSuccess?: () => void,
+        onFailure?: () => void
+    ): void => {
+        settLagrerVilkår(true);
+        request<IRestPersonResultat, IBehandling>({
+            method: 'PUT',
+            url: `/familie-ks-sak/api/vilkaarsvurdering/${behandlingId}/${vilkårId}`,
+            data: restPersonResultat,
+        })
+            .then((response: Ressurs<IBehandling>) => {
+                settLagrerVilkår(false);
+                if (response.status === RessursStatus.SUKSESS) {
```

Comment:
Hva skjer ved response.status !== SUKSESS ?

Reply:
settÅpenBehandling forventer en `Ressurs<IBehandling>`. Kunne vi bare satt den direkte, uavhengig av statusen?

Reply:
Vi burde nok håndtere ```response.status !== SUKSESS``` ihvertfall 👍 Sette en feilmelding som kan vises blandt annet. Grunnen til jeg foreløpig ikke har satt respons direkte er for å unngå å skrive over det som ligger i frontend dersom det feiler. Vil heller la saksbehandler kunne rette opp det som er feil enn å potensielt tømme skjemaer osv